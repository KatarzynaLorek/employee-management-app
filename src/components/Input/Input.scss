.input {
  &_label {
    display: block;
    width: 200px;
    font-size: 16px;
  }
  &_select {
    margin-top: 5px;
    width: 200px;
    height: 30px;
    font-size: 16px;
    line-height: 30px;
  }
}

@media (max-width: 1000px) {
  .input {
    &_select {
      height: 40px;
      line-height: 40px;
    }
  }
}

@media (max-width: 450px) {
  .input {
    &_select {
      width: 100%;
      max-width: 450px;
    }
  }
}
