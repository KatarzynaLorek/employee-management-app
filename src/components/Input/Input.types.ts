export enum InputVariant {
  sortingType = 'sortingType',
  chartType = 'chartType',
}

export interface IInput<T> {
  variant: InputVariant;
  options: string[];
  label: string;
  containerClass: string;
  handleChange: (type: T) => void;
}
