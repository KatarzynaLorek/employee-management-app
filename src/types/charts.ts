export type SortType = 'department' | 'position';
export type ChartType = 'bar' | 'doughnut';
export type DepartmentType = 'accountancy' | 'IT' | 'sales' | 'marketing';
export type PositionType = 'junior' | 'regular' | 'senior' | 'manager';
