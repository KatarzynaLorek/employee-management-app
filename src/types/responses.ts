export interface IResponseObject {
  id: string;
  firstName: string;
  lastName: string;
  department: 'accountancy' | 'IT' | 'sales' | 'marketing';
  position: 'junior' | 'regular' | 'senior' | 'manager';
}
