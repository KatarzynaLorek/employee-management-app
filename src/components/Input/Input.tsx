import React from 'react';
import { ChartType, SortType } from '../../types/charts';
import './Input.scss';

interface ICommonInputProps {
  type: string;
  options: string[];
  label: string;
  containerClass: string;
}

interface IChartInputProps extends ICommonInputProps {
  chartType: true;
  changeType: (type: ChartType) => void;
}

interface ISortInputProps extends ICommonInputProps {
  chartType?: false;
  changeType: (type: SortType) => void;
}

type IInputProps = IChartInputProps | ISortInputProps;

const Input = (props: IInputProps): JSX.Element => {
  return (
    <div className={props.containerClass}>
      <label className="input_label" htmlFor={props.type}>
        {props.label}
      </label>
      <select
        className="input_select"
        name={props.type}
        onChange={(e) => {
          props.chartType
            ? props.changeType(e.target.value as ChartType)
            : props.changeType(e.target.value as SortType);
        }}
      >
        {props.options.map((option) => (
          <option key={option} value={option}>
            {option}
          </option>
        ))}
      </select>
    </div>
  );
};

export default Input;
