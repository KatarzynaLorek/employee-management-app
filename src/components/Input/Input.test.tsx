import React from 'react';
import Input from './Input';
import { IResponseObject } from '../../types/responses';
import { render, fireEvent, waitFor } from '@testing-library/react';

const mockProps = {
  type: 'mockType',
  options: ['mockOption1', 'mockOption2'],
  label: 'mockLabel',
  containerClass: 'mockContainerClass',
  handleChange: jest.fn,
  isChartType: true,
};

describe('Input Element', () => {
  it('renders properly without isChartType props', () => {
    const container = render(
      <Input
        type={mockProps.type}
        options={mockProps.options}
        label={mockProps.label}
        containerClass={mockProps.containerClass}
        handleChange={mockProps.handleChange}
      />,
    );
    expect(container.baseElement).toMatchSnapshot();
  });
  it('renders properly with isChartType props', () => {
    const container = render(
      <Input
        type={mockProps.type}
        options={mockProps.options}
        label={mockProps.label}
        containerClass={mockProps.containerClass}
        handleChange={mockProps.handleChange}
        isChartType
      />,
    );
    expect(container.baseElement).toMatchSnapshot();
  });
});
