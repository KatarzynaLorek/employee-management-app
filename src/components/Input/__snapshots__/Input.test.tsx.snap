// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Element renders properly with isChartType props 1`] = `
<body>
  <div>
    <div
      class="mockContainerClass"
    >
      <label
        class="input_label"
        for="mockType"
      >
        mockLabel
      </label>
      <select
        class="input_select"
        name="mockType"
      >
        <option
          value="mockOption1"
        >
          mockOption1
        </option>
        <option
          value="mockOption2"
        >
          mockOption2
        </option>
      </select>
    </div>
  </div>
</body>
`;

exports[`Input Element renders properly without isChartType props 1`] = `
<body>
  <div>
    <div
      class="mockContainerClass"
    >
      <label
        class="input_label"
        for="mockType"
      >
        mockLabel
      </label>
      <select
        class="input_select"
        name="mockType"
      >
        <option
          value="mockOption1"
        >
          mockOption1
        </option>
        <option
          value="mockOption2"
        >
          mockOption2
        </option>
      </select>
    </div>
  </div>
</body>
`;
